src/Microsoft.TemplateEngine.Utils/JsonHelpers.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions.Json;
+
+namespace Microsoft.TemplateEngine.Utils
+{
+    public static class JsonHelpers
+    {
+        public static (string, Action<IJsonToken>) CreateArrayValueExtractor<TType>(string key, List<TType> target)
+        {
+            return (key,
+                (token) =>
+                {
+                    if (token is IJsonArray tokenArray)
+                    {
+                        foreach (IJsonValue element in tokenArray)
+                        {
+                            target.Add((TType)element.Value);
+                        }
+                    }
+                }
+            );
+        }
+
+        public static (string, Action<IJsonToken>)[] CreateStringKeyDictionaryExtractor<TValue>(IJsonObject token, IDictionary<string, TValue> target)
+        {
+            List<(string, Action<IJsonToken>)> extractorList = new List<(string, Action<IJsonToken>)>();
```

Comment:
This could be made to be an array if we pre-collapse the set of property names & iterate over the collapsed set. Alternatively, modifying the abstraction over object could let us get away with only having to enumerate the underlying store once

Reply:
Changed to constructing the extractors directly as an array.